blocks/api/parser.js

```diff
@@ -88,6 +53,69 @@ export function asType( value, type ) {
 }
 
 /**
+ * Returns an hpq matcher given a source object
+ *
+ * @param  {Object}   sourceConfig Attribute Source object
+ * @return {Function}              hpq Matcher
+ */
+export function matcherFromSource( sourceConfig ) {
+	switch ( sourceConfig.source ) {
+		case 'attribute':
+			return attr( sourceConfig.selector, sourceConfig.attribute );
+		case 'property':
+			return prop( sourceConfig.selector, sourceConfig.property );
+		case 'html':
+			return html( sourceConfig.selector );
+		case 'text':
+			return text( sourceConfig.selector );
+		case 'children':
+			return children( sourceConfig.selector );
+		case 'node':
+			return node( sourceConfig.selector );
+		case 'query':
+			return query( sourceConfig.selector, matcherFromSource( sourceConfig.query ) );
+		case 'object':
+			return keys( sourceConfig.object ).reduce( ( memo, key ) => {
+				memo[ key ] = matcherFromSource( sourceConfig.object[ key ] );
+				return memo;
+			}, {} );
+		default:
+			// eslint-disable-next-line no-console
+			console.error( `Unkown source type "${ sourceConfig.source }"` );
+	}
+}
+
+/**
+ * Given an attribute key, an attribute's schema, a block's raw content and the commentAttributes
+ * returns the attribute value depending on its source definition of the given attribute key
+ *
+ * @param  {string} attributeKey        Attribute key
+ * @param  {Object} attributeSchema     Attribute's schema
+ * @param  {string} rawContent          Block's raw content
+ * @param  {Object} commentAttributes   Block's comment attributes
+ *
+ * @return {*}                          Attribute value
+ */
+export function getBlockAttribute( attributeKey, attributeSchema, rawContent, commentAttributes ) {
+	let value;
+	switch ( attributeSchema.source ) {
+		case 'meta':
```

Comment:
Our approach for defining other attribute sources is a bit awkward, I think, in that we have to explicitly bypass the getter here to allow it to be handled elsewhere later. Would like if these strategies were more consolidated. cc @mcsf